sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Listeners/QueueTargetScaler.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Storage.Queues;
+using Microsoft.Azure.WebJobs.Extensions.Storage.Common.Listeners;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common.Listeners
+{
+    /// <summary>
+    /// Provides queue length metrics for the <see cref="QueueTargetScaler"/>.
+    /// </summary>
+    internal sealed class QueueTargetScaler : ITargetScaler
+    {
+        private readonly string _functionId;
+        private readonly QueueMetricsProvider _queueMetricsProvider;
+        private readonly TargetScalerDescriptor _targetScalerDescriptor;
+        private QueuesOptions _options;
+        private readonly ILogger _logger;
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        public TargetScalerDescriptor TargetScalerDescriptor => _targetScalerDescriptor;
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="functionId"></param>
+        /// <param name="queueClient"></param>
+        /// <param name="options"></param>
+        /// <param name="loggerFactory"></param>
+        internal QueueTargetScaler(string functionId, QueueClient queueClient, QueuesOptions options, ILoggerFactory loggerFactory)
+        {
+            _functionId = functionId;
+            _queueMetricsProvider = new QueueMetricsProvider(queueClient,loggerFactory);
+            _targetScalerDescriptor = new TargetScalerDescriptor(functionId);
+            _options = options;
+            _logger = loggerFactory.CreateLogger<QueueTargetScaler>();
+        }
+
+        /// <summary>
+        /// Makes a target scale decision based on most recent metrics for the specified queue.
+        /// </summary>
+        /// <param name="context">The TargetScalerContext, which contains the InstanceConcurrency, or the targetMetric used in target based scaling.</param>
+        /// <returns>Returns a TargetScalerResult with a TargetWorkerCount.</returns>
+        public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            var metrics = await _queueMetricsProvider.GetMetricsAsync().ConfigureAwait(false);
+            return GetScaleResultInternal(context, metrics);
+        }
+
+        internal TargetScalerResult GetScaleResultInternal(TargetScalerContext context, QueueTriggerMetrics metrics)
+        {
+            int concurrency = !context.InstanceConcurrency.HasValue ? _options.BatchSize : context.InstanceConcurrency.Value;
```

Comment:
do we want throw ArgumentOutOfRangeException if concurrency <1:
https://github.com/Azure/azure-sdk-for-net/pull/31367/files#diff-248e4fb51e016f725b801cfe278780e23a8a07cd7500a1efdccfaf1983d64378R94

Reply:
added this exception